### ISSUE TITLE
AttributeError: Module `numpy` has no attribute `asscalar`

### ISSUE THREAD

**Issue by mn5hk** [astropy/astropy#15075]
### Description

Hi, it appears some part of the code is using depreciated numpy codes. Request it to be fixed.

![image](https://github.com/astropy/astropy/assets/65601810/d1c0a4d2-b4ed-4e8c-84e5-6953e981b3a1)

Thank you,
AS

### How to Reproduce

1. Get package from '...'
2. Then run '...'
3. An error occurs.

```python
# Put your Python code snippet here.
```

### Versions

import platform; print(platform.platform())
import sys; print("Python", sys.version)
import astropy; print("astropy", astropy.__version__)
import numpy; print("Numpy", numpy.__version__)
import erfa; print("pyerfa", erfa.__version__)
import scipy; print("Scipy", scipy.__version__)
import matplotlib; print("Matplotlib", matplotlib.__version__)

**Comment by WilliamJamieson** [comment]
In order to diagnose this issue we will need you to provide the request versions output. That is please add the output of:
```
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import astropy; print("astropy", astropy.__version__)
import numpy; print("Numpy", numpy.__version__)
import erfa; print("pyerfa", erfa.__version__)
import scipy; print("Scipy", scipy.__version__)
import matplotlib; print("Matplotlib", matplotlib.__version__)
```

You can paste this into your Python repl, notebook, ipython repl, or a Python file and then run it to get the result.

**Comment by saimn** [comment]
You must have a quite old Astropy version, this was fixed in 5.0 (#12535).

**Comment by eerovaher** [comment]
As pointed out already the problem occurs only when running `numpy` >= 1.23 together with `astropy`  < 5.0.1. Later versions of `astropy` don't have this problem and older versions are not supported.
